packages/SwingSet/src/devices/mailbox.js

```diff
@@ -72,12 +72,36 @@ import Nat from '@agoric/nat';
 // replace it with one that tracks which parts of the state have been
 // modified, to build more efficient Merkle proofs.
 
-export function buildMailboxStateMap() {
-  const state = harden(new Map());
+export function importMailbox(data, inout = {}) {
+  const outbox = new Map();
+  data.outbox.forEach(m => {
+    outbox.set(Nat(m[0]), m[1]);
```

Comment:
Huh, this reminds me how challenging it will be to maintain determinism in the face of antagonistic code which is trying to examine every conceivable input for subtle variations in behavior, like the insertion order of maps. We're fine here (assuming `importMailbox` is fed the same data that came out of `exportMailbox`, it will always be sorted, so the new `outbox` will have a consistent insertion order), and for normal vats we're probably safe (the `args` they get are created by JSON.parse, which at least has one obvious order to pick). But we'll probably need to do a determinism audit to find places where, I dunno, maybe we provide an error object or the new `vatParameters` or something, and while the contents are always exactly the same, some incarnations of the process build the object differently, allowing sufficiently determined code to behave nondeterministically.

I think we need to guard access to objects that might have hidden nondeterminism, and only allow vetted code to examine them. But that sounds a lot more difficult to search for than our usual "objects with significant authority" criteria.